### PR TITLE
36 other auth solutions for dashboards

### LINKED ISSUE #36: Other Auth Solutions for Dashboards
Currently, the dashboards are accessed through BasicAuth. It is fine but it means sharing a password so it would be better to use something like GitHub account based authorization solutions.

### PR DESCRIPTION
**Pull request by hasanheroglu**
Closes #36.

Here, I added a forward auth mechanism. Therefore traefik forwards requests for prometheus, cadvisor and traefik to GitHub authentication. We need to set an environment variable (WHITELISTED_EMAILS) to whitelist emails. Also, we need to create a GitHub OAuth application and use its configuration to successfully log in.

Grafana and Portainer use their own authentication mechanisms and we need to configure them using their UIs. For Grafana, it is possible to give permissions depending on the user's GitHub data. A simple role attribute path would be `[email==<USER_EMAIL>] && 'Admin' || 'Viewer'` to give specific USER_EMAIL admin permissions and all other viewer permissions. Check [Grafana documentation](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/github/) for more details. Furthermore, it is possible to create teams and limit access to dashboards to certain teams and users.

### CODE REVIEW

File: docker-compose-infra.yml [hunk 1]
```diff
@@ -37,6 +35,24 @@ services:
         reservations:
           cpus: '0.05'
           memory: 50M
+  test-things-auth:
+    image: thomseddon/traefik-forward-auth:2-arm64
```
Comment (egekorkan) [inline docker-compose-infra.yml:39]: Will need to check if we can use the arm image

File: .env [hunk 2]
```diff
@@ -15,9 +15,17 @@ TRAEFIK_DASHBOARD_PORT_OUT=8080
...
+OAUTH_SECRET=
+GITHUB_AUTH_URL=https://github.com/login/oauth/authorize
+GITHUB_TOKEN_URL=https://github.com/login/oauth/access_token
+GITHUB_USER_URL=https://api.github.com/user
+GITHUB_CLIENT_ID=
+GITHUB_CLIENT_SECRET=
+WHITELISTED_EMAILS=
 
+# SERVICE PORTS
+FORWARD_AUTH_PORT=4181
```
Comment (egekorkan) [inline .env:28]: This port is not open afaik

**Review by egekorkan — COMMENTED** [review]
Overall looks good. The comments above are due to production environment

File: docker-compose-infra.yml (hunk 1 above)
Comment (hasanheroglu) [reply docker-compose-infra.yml:39]: This repo for forward auth is not very well managed, even though this says arm64 all the images in Docker Hub are amd64. I can rename this.

File: .env (hunk 2 above)
Comment (hasanheroglu) [reply .env:28]: I assume we don't need to expose this port. It is just a port number forward-auth is running in the container.

**Pull request merged**